microovn/node/node.go

```diff
@@ -216,3 +221,89 @@ func ServiceWarnings(ctx context.Context, s state.State) (types.WarningSet, erro
 	}
 	return output, nil
 }
+
+// StartCentral safely starts the central service and its child service while also generating certificates
+func StartCentral(ctx context.Context, s state.State) error {
+	// Generate certificate for OVN Central services
+	err := certificates.GenerateNewServiceCertificate(ctx, s, "ovnnb", certificates.CertificateTypeServer)
+	if err != nil {
+		return fmt.Errorf("failed to generate TLS certificate for ovnnb service")
+	}
+	err = certificates.GenerateNewServiceCertificate(ctx, s, "ovnsb", certificates.CertificateTypeServer)
+	if err != nil {
+		return fmt.Errorf("failed to generate TLS certificate for ovnsb service")
+	}
+	err = certificates.GenerateNewServiceCertificate(ctx, s, "ovn-northd", certificates.CertificateTypeServer)
+	if err != nil {
+		return fmt.Errorf("failed to generate TLS certificate for ovn-northd service")
+	}
+
+	err = snap.Start("ovn-ovsdb-server-nb", true)
+	if err != nil {
+		return fmt.Errorf("Failed to start OVN NB: %w", err)
+	}
+
+	err = snap.Start("ovn-ovsdb-server-sb", true)
+	if err != nil {
+		return fmt.Errorf("Failed to start OVN SB: %w", err)
+	}
+
+	err = snap.Start("ovn-northd", true)
+	if err != nil {
+		return fmt.Errorf("Failed to start OVN northd: %w", err)
+	}
+	return nil
+}
+
+// StopCentral safely stops the central service and its child service
```

Comment:
Similar to above, this function is doing more than stopping the `central` service, which may come as a surprise to someone making use of the function in the future (including future you).

Would it make more sense to call it `LeaveCentral` or something like that?

Reply:
This appears to be addressed in commit "Better (enabling|disabling) of central services" later in this series.